### PR TITLE
[BUG] -1 to the power of pd.NA was returning -1

### LINKED ISSUE #30956: BUG: rpow with -1 and pd.NA should return pd.NA
In https://github.com/pandas-dev/pandas/pull/30097, we added special cases for 1 and -1 not propagating pd.NA:

```
In [59]: 1 ** pd.NA  
Out[59]: 1

In [60]: (-1) ** pd.NA
Out[60]: -1
```

But for -1 this should probably be pd.NA as result in the end, see https://mail.python.org/pipermail/pandas-dev/2020-January/001174.html

```
In [58]: (-1) ** np.array([2, 3, 2.5]) 
/home/joris/miniconda3/envs/dev/bin/ipython:1: RuntimeWarning: invalid value encountered in power
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[58]: array([ 1., -1., nan])
```

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] closes #30956 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
> ##[error]The job running on agent Azure Pipelines 9 ran longer than the maximum time of 60 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134

I don't know what to make of this error, sorry

File: doc/source/user_guide/missing_data.rst [hunk 1]
```diff
@@ -831,7 +831,7 @@ Operation        Result
 ================ ======
 ``pd.NA ** 0``   0
 ``1 ** pd.NA``   1
-``-1 ** pd.NA``  -1
+``-1 ** pd.NA``  pd.NA
```
Comment (TomAugspurger) [inline doc/source/user_guide/missing_data.rst]: I think just remove this line. The sentence above describes this as "the result is known, even when one of the
operands is `NA`"

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Comment by simonjayhawkins** [comment]
@jorisvandenbossche in these situations is a whatsnew note needed?

**Comment by jorisvandenbossche** [comment]
That's not needed in this case, since it is a new feature, not a bugfix compared to a previous release.

**Comment by simonjayhawkins** [comment]
although it's a user reported bug from the rc release. could be worth considering in future having a short section in the whatsnew listing changes since the rc.

**Comment by TomAugspurger** [comment]
I think we're OK without one.

**Comment by TomAugspurger** [comment]
Thanks @MarcoGorelli!

**Comment by jorisvandenbossche** [comment]
@MeeseeksDev backport 1.0.x

**Comment by jorisvandenbossche** [comment]
For some reason this got not backported (we will need to check it a bit ...)

**Pull request merged**